flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1264,6 +1273,483 @@ def test_all_transitions(self, desired_state, discovered_state):
+                    dataset_id=ScenarioMixin.DATASET_ID,
+                    metadata=ScenarioMixin.METADATA,
+                ),
+            ],
+        )
+
+    def test_deleted_attached_manifest_dataset(self):
+        """
+        If there is a non-manifest dataset that is deleted on this
+        node, the corresponding dataset has a desired state of
+        ``DELETED``.
+
+        This is only necessary until we can delete non-manifest datasets
+        (FLOC-1772).
+        """
+        assert_desired_datasets(
+            self, self.deployer,
+            desired_manifestations=[
+                ScenarioMixin.MANIFESTATION.transform(
+                    ["dataset", "deleted"], True,
+                    ['dataset', 'metadata'], ScenarioMixin.METADATA,
+                ),
+            ],
+            # XXX FLOC-1772 We only need to do this to handle
+            # deleting unmounted datasets
+            local_datasets=[
+                DiscoveredDataset(
+                    dataset_id=ScenarioMixin.DATASET_ID,
+                    blockdevice_id=ScenarioMixin.BLOCKDEVICE_ID,
+                    state=DatasetStates.ATTACHED,
+                    maximum_size=int(REALISTIC_BLOCKDEVICE_SIZE.bytes),
+                    device_path=FilePath('/dev/xvdf'),
+                ),
+            ],
+            expected_datasets=[
+                DesiredDataset(
+                    state=DatasetStates.DELETED,
+                    dataset_id=ScenarioMixin.DATASET_ID,
+                    metadata=ScenarioMixin.METADATA,
+                ),
+            ],
+        )
+
+    def test_deleted_non_manifest_dataset(self):
+        """
+        If there is a manfestation on this node that is deleted,
+        the corresponding dataset has a desired state of ``DELETED``.
+        """
+        assert_desired_datasets(
+            self, self.deployer,
+            desired_manifestations=[
+                ScenarioMixin.MANIFESTATION.transform(
+                    ["dataset", "deleted"], True,
+                    ['dataset', 'metadata'], ScenarioMixin.METADATA,
+                ),
+            ],
+            local_datasets=[
+                DiscoveredDataset(
+                    dataset_id=ScenarioMixin.DATASET_ID,
+                    blockdevice_id=ScenarioMixin.BLOCKDEVICE_ID,
+                    state=DatasetStates.NON_MANIFEST,
+                    maximum_size=int(REALISTIC_BLOCKDEVICE_SIZE.bytes),
+                ),
+            ],
+            expected_datasets=[
+                DesiredDataset(
+                    state=DatasetStates.DELETED,
+                    dataset_id=ScenarioMixin.DATASET_ID,
+                    metadata=ScenarioMixin.METADATA,
+                ),
+            ],
+        )
+
+    def test_leased_mounted_manifestaion(self):
+        """
+        If there is a lease for a mounted dataset present on node,
+        there is a corresponding dataset that has a desired state of
+        ``MOUNTED``.
```

Comment:
add something like "even if the configuration does not mention the dataset, since the lease keeps it from being moved."